### DISCUSSION TITLE
Can we generate sapn for all internal calling methods from service

### DISCUSSION THREAD

**Discussion by ychaydhari** [open-telemetry/opentelemetry-java#4803] [Q&A]
Hello,
I have created one controller and called one service method from that controller. This service calls multiple methods But I'm getting traced up to 2 spans only. 

After adding the WithSpan annotation I got span for that particular service method but not for all calling methods. Can we generate a span for all calling methods?  So it's better to find method latency or if any particular method failed.

![image](https://user-images.githubusercontent.com/106724177/193019510-5da94033-f5a3-46bb-9cd0-bd5e7f57c708.png)

**Comment by jack-berg** [comment]
The `@WithSpan` annotation creates new spans when the annotated method is invoked. It cannot be used to add spans to calling methods.

**Reply by ychaydhari** [reply]
Do we have any way to generate a span on all calling methods from that service? instead of adding @WithSpan on all the methods
